src/main.rs

```diff
@@ -234,37 +242,50 @@ async fn handle_request(
         }
     };
 
-    let bytes = match hyper::body::to_bytes(body).await {
-        Ok(body) => body.to_vec(),
+    let p = path_name(&path);
+
+    let header_sender = match ratelimiter.ticket(path).await {
+        Ok(sender) => sender,
         Err(e) => {
-            error!("Failed to receive incoming request body: {:?}", e);
-            return Err(RequestError::ChunkingRequest { source: e });
+            error!("Failed to receive ticket for ratelimiting: {:?}", e);
+            return Err(RequestError::AcquiringTicket { source: e });
         }
     };
 
-    let path_and_query = if let Some(v) = uri.path_and_query() {
-        v.as_str().replace(&api_url, "")
-    } else {
-        error!("No path in URI: {:?}", uri);
-        return Err(RequestError::NoPath { uri });
-    };
-    let p = path_name(&path);
-    let raw_request = RequestBuilder::raw(method, path, path_and_query)
-        .body(bytes)
-        .headers(headers.into_iter().filter_map(|(k, v)| k.map(|h| (h, v))))
-        .build();
+    parts
+        .headers
+        .insert(AUTHORIZATION, token.try_into().unwrap());
```

Comment:
Is this still needed?

Reply:
yeah it is needed since token may not be provided in which case the default is used